### PR TITLE
let dmd.root.speller.spellerX use a seed array

### PR DESCRIPTION
**Pull request by Herringway**
Second verse, same as the first.

### CODE REVIEW

File: src/dmd/root/speller.d [hunk 1]
```diff
@@ -95,83 +95,82 @@ private auto spellerY(alias dg)(const(char)[] seed, size_t index, ref int cost)
...
 {
-    if (!seedlen)
+    if (!seed.length)
         return null;
     char[30] tmp;
-    char* buf;
-    if (seedlen <= tmp.sizeof - 2)
-        buf = tmp.ptr;
+    char[] buf;
+    if (seed.length <= tmp.sizeof - 1)
```
Comment (RazvanN7) [inline src/dmd/root/speller.d]: Why is this `-1` as opposed to `-2`?

File: src/dmd/root/speller.d [hunk 2]
```diff
@@ -95,83 +95,82 @@ private auto spellerY(alias dg)(const(char)[] seed, size_t index, ref int cost)
...
-        if (!buf)
-            return null; // no matches
+        buf = (cast(char*)alloca(seed.length + 1))[0 .. seed.length + 1]; // leave space for extra char
     }
     int cost = int.max, ncost;
     searchFunctionType!dg p = null, np;
     /* Deletions */
-    buf[0 .. seedlen] = seed[1 .. seedlen + 1];
-    for (size_t i = 0; i < seedlen; i++)
+    buf[0 .. seed.length - 1] = seed[1 .. $];
```
Comment (RazvanN7) [inline src/dmd/root/speller.d]: Why the -1?

File: src/dmd/root/speller.d (hunk 1 above)
Comment (Herringway) [reply src/dmd/root/speller.d]: Space for the terminating null. No longer needed.

File: src/dmd/root/speller.d (hunk 2 above)
Comment (Herringway) [reply src/dmd/root/speller.d]: No longer need to copy a terminating null.

**Review by RazvanN7 — APPROVED** [review]

**Pull request merged**